src/udev/udev-node.c

```diff
@@ -232,7 +228,39 @@ static int link_update(sd_device *dev, const char *slink, bool add) {
                                 r = -errno;
                 } while (r == -ENOENT);
 
-        return r;
+        for (i = 0; i < LINK_UPDATE_MAX_RETRIES; i++) {
```

Comment:
in newer code we declare variables in the for loop, i.e. write like this:

```c
for (unsigned i = 0; i < …
```

and remove the `int i` from the top

Reply:
Actually, i is used to figure out whether we return `-ELOOP`

Reply:
good point